protocols/v2/const-sv2/src/lib.rs

```diff
@@ -1,114 +1,203 @@
-//! Central repository for all the sv2 constants
+//! This crate serves as the central repository for all constants used in the SV2 protocol.
+//! These constants are essential for message framing, encryption, and protocol-specific
+//! identifiers across various SV2 components, including Mining, Job Declaration, and Template Distribution protocols.
+//!
+//! It also includes definitions for key encryption settings and message types, ensuring consistency and accuracy
+//! when working with SV2's binary protocol. These constants are used throughout the system to ensure that SV2
+//! messages are formatted and interpreted correctly.
+//!
+//! ### Discriminants for Stratum V2 (sub)protocols
+//! Discriminants are unique identifiers used to distinguish between different Stratum V2 (sub)protocols.
+//! Each protocol within the SV2 ecosystem has a specific discriminant value that indicates its type,
+//! enabling the correct interpretation and handling of messages.
+//! These discriminants ensure that messages are processed by the appropriate protocol handlers,
+//! thereby facilitating seamless communication across different components of the SV2 architecture.
+//! More info at <https://github.com/stratum-mining/sv2-spec/blob/main/03-Protocol-Overview.md#3-protocol-overview>
```

Comment:
we should leverage markdown rendering for links, plaintext hurts reader eyes
```suggestion
//! More info can be found [on Chapter 03 of the Stratum V2 spec](https://github.com/stratum-mining/sv2-spec/blob/main/03-Protocol-Overview.md#3-protocol-overview).
```

Reply:
Done here d84c53cc91b97fae08c5b28f90ecd16a7cea9254